src/Views/EditDog/EditDog.js

```diff
@@ -0,0 +1,68 @@
+import React, { useEffect, useState } from 'react';
+import { useHistory, useParams } from 'react-router-dom';
+import DogForm from '../../Comps/DogForm';
+import { editDog, fetchDogById } from '../../services/fetchdogs';
+
+export default function EditPage() {
+  
+  const [img, setImg] = useState('');
+  const [Name, setName] = useState('');
+  const [Age, setAge] = useState('');
+  const [Breed, setBreed] = useState('');
+  const [Desc, setDesc] = useState('');
+  const [error, setError] = useState('');
+  const [message, setMessage] = useState('');
+  
+
+  const history = useHistory(); 
+  const params = useParams();
+  const id = params.id;
+
+  useEffect(() => {
+    const fetchDog = async () => {
+      const data = await fetchDogById(id);
+      setName(data.Name);
+      setAge(data.Age);
+      setBreed(data.Breed);
+      setDesc(data.Desc);
+      setError(data.error);
+      setImg(data.img);
+    };
+    fetchDog();
+  }, [id]);
+
+  const handleSubmit = async (e) => {
+
+    e.preventDefault();
+    try {
+      await editDog({ id, Name, Desc, Breed, img, Age });
+      setMessage('Edit Good');
+      history.push(`/dogs/${id}`);
```

Comment:
Here we will want to call a setTimeout function so the user can see the message before being redirected to home.